Fix sched event deletion - recordings endpoint

Alaschinger opened the pull request:
Scheduled events will be completly deleted instead of just removing the capture agent ID when using the DELETE /recordings/ endpoint. This fixes #4273 

### Your pull request should…

* [ ] have a concise title
* [ ] [close an accompanying issue](https://help.github.com/en/articles/closing-issues-using-keywords) if one exists
* [ ] [be against the correct branch](https://docs.opencast.org/develop/developer/development-process#acceptance-criteria-for-patches-in-different-versions)
* [ ] include migration scripts and documentation, if appropriate
* [ ] pass automated tests
* [ ] have a clean commit history
* [ ] [have proper commit messages (title and body) for all commits](https://medium.com/@steveamaza/e028865e5791)

Alaschinger committed:
fix sched event deletion - recordings endpoint

Scheduled events will be completly deleted instead of just removing the capture agent when using the DELETE /recordings/ endpoint.

geichelberger commented on modules/scheduler-impl/src/main/java/org/opencastproject/scheduler/impl/SchedulerServiceImpl.java:
```diff
@@ -1523,6 +1523,7 @@ private void removeSchedulingInfoFromIndex(String mediaPackageId) {
 
     try {
       index.addOrUpdateEvent(mediaPackageId, updateFunction, orgId, user);
+      index.deleteEvent(mediaPackageId, orgId);
```
Is only deleting the event not sufficient? Why does it have to be updated and deleted afterward?

Alaschinger replied:
In theory it is. I really don't know why there was a step specifically keeping the event with an empty capture agent ID.

Arnei replied:
As far as I can tell, updating the event before deletion has no beneficial side-effects. Therefore I went ahead and removed the update related code. Do you agree with this change @geichelberger ?

Arnei committed:
Remove unnecessary update before delete in ES

The code removed by this PR was previously used to "remove" the event.
Since at this point in time no one knows anymore why that was done,
the code is deleted and removing is left to the call to `index.deleteEvent`.
